File: Java/Easy/Accepted/truncate-sentence_3.java
class Solution {
    public String truncateSentence(String s, int k) {
        
        int count = 0;
        String result = "";
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == ' ') {
                count++;
                if (count == k) {
                    break;
                }
            }
            result += s.charAt(i);
        }
        return result;
    }
}

File: Rust/Easy/Accepted/truncate-sentence_3.rs
impl Solution {
    pub fn truncate_sentence(s: String, k: i32) -> String {
        
        let mut res = String::new();
        let mut count = 0;
        for c in s.chars() {
            if c == ' ' {
                count += 1;
            }
            if count == k {
                break;
            }
            res.push(c);
        }
        res
    }
}